spark/src/main/scala/ai/chronon/spark/Analyzer.scala

```diff
@@ -407,8 +407,38 @@ class Analyzer(tableUtils: TableUtils,
         )
       }
     }
+    // Derive the join online fetching output schema with metadata
+    val aggMetadata: ListBuffer[AggregationMetadata] = if (joinConf.hasDerivations) {
+      val keyColumns: List[String] = joinConf.joinParts.toScala
+        .flatMap(joinPart => {
+          val keyCols: Seq[String] = joinPart.groupBy.keyColumns.toScala
+          if (joinPart.keyMapping == null || joinPart.keyMapping.isEmpty) keyCols
+          else {
+            keyCols.map(key => joinPart.keyMapping.getOrDefault(key, key))
```

Comment:
@yuli-han keyMapping is a mapping from left_key to right_key, but here you are doing the reverse

Reply:
Let's include this in a unit test

Reply:
good catch

Reply:
thanks for catching! Updated the code and also added a unit test to cover it. Also tested locally using a manually created testing config and attached the result in the above testing result doc.